test/Kestrel.Transport.FunctionalTests/Http2/ChromeTests.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#if NETCOREAPP2_2
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
+using Microsoft.AspNetCore.Server.Kestrel.FunctionalTests;
+using Microsoft.AspNetCore.Testing;
+using Microsoft.AspNetCore.Testing.xunit;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Testing;
+using Xunit;
+
+namespace Interop.FunctionalTests
+{
+    [OSSkipCondition(OperatingSystems.MacOSX, SkipReason = "Missing SslStream ALPN support: https://github.com/dotnet/corefx/issues/30492")]
+    [MinimumOSVersion(OperatingSystems.Windows, WindowsVersions.Win81, SkipReason = "Missing Windows ALPN support: https://en.wikipedia.org/wiki/Application-Layer_Protocol_Negotiation#Support")]
+    [SkipIfChromeUnavailable]
+    public class ChromeTests : LoggedTest
+    {
+        private static readonly string _chromeArgs = "--headless --disable-gpu --allow-insecure-localhost --enable-logging --dump-dom --virtual-time-budget=10000";
+
+        [ConditionalTheory]
+        [InlineData("", "Interop HTTP/2 GET")]
+        [InlineData("?TestMethod=POST", "Interop HTTP/2 POST")]
+        public async Task Http2(string requestSuffix, string expectedResponse)
+        {
+            using (var server = new TestServer(async context =>
+            {
+                if (string.Equals(context.Request.Query["TestMethod"], "POST", StringComparison.OrdinalIgnoreCase))
+                {
+                    await context.Response.WriteAsync(
+@"<!DOCTYPE html>
+<html>
+ <head>
+   <script type=""text/javascript"">
+    function dosubmit() { document.forms[0].submit(); }
+   </script>
+ </head>
+ <body onload=""dosubmit();"">
+   <form action=""/"" method=""POST"" accept-charset=""utf-8"">
+   </form>
+ </body>
+</html>");
+                }
+                else
+                {
+                    await context.Response.WriteAsync($"Interop {context.Request.Protocol} {context.Request.Method}");
+                }
+            },
+            new TestServiceContext(LoggerFactory),
+            options => options.Listen(IPAddress.Loopback, 0, listenOptions =>
+            {
+                listenOptions.Protocols = HttpProtocols.Http2;
+                listenOptions.UseHttps(TestResources.GetTestCertificate());
+            })))
+            {
+                var netLogPath = Path.Combine(ResolvedLogOutputDirectory, $"{ResolvedTestMethodName}.nl.json");
```

Comment:
extract this boilerplate to a helper method or class.

Reply:
I don't think that's needed since there's only one test right now. If we had more tests I would extract them.

Reply:
All of this busywork is distracting from what the test is actually trying to test.